src/plugins/Wallet/wallet.ts

```diff
@@ -211,5 +211,16 @@ export async function onWalletERC20TokenBalanceUpdated(address: string, tokenAdd
 async function getWalletByAddress(t: IDBPSafeTransaction<WalletDB, ['Wallet'], 'readonly'>, address: string) {
     const rec = await t.objectStore('Wallet').get(address)
     assert(rec)
+    if (rec.eth_balance) {
+        Object.setPrototypeOf(rec.eth_balance, BigNumber.prototype)
```

Comment:
Store a string like "1000000000000000000000000000000000" into the database and read it like `new BigNumber(rec.eth_balance)`. If you choose to drop the prototype and store it directly, we will depend on the internal data structure of BigNumber forever.